psi4/driver/procrouting/dft_funcs/superfuncs.py

```diff
@@ -182,6 +187,23 @@ def build_superfunctional(name, restricted):
     # Lock and unlock the functional
     sup[0].set_lock(False)
 
+    # pre-defined -NL dashparam first, then special dft_disp_parm option.
+    # not applicable for XC generating function
+    if not hasattr(name, '__call__'):
+        if (any(name.lower().endswith(al) for al in dftd3.full_dash_keys)):
+            dashparam = [x for x in dftd3.full_dash_keys if name.endswith(x)]
+            if (dashparam[0] == 'nl'):
+                base_name = name.replace('-' + dashparam[0], '')
+                sup[0].set_vv10_b(dftd3.dashcoeff['nl'][base_name.lower()]['b'])
```

Comment:
In my opinion, this should be handled in the EmpiricalDispersion class - we pass the `"dispersion" : {"type": "nl", "params": {"b": B, "c": C}, "citation": "STR"}` in, analogously to all other functionals, and the class figures it out. The `if`-`else` on lines 75-106 of empirical_dispersion.py will need a third fork for `nl`.